src/Console/Output/Progress/ProgressOutputInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output\Progress;
+
+use PhpCsFixer\FixerFileProcessedEvent;
+
+/**
+ * @internal
+ */
+interface ProgressOutputInterface
+{
+    public const OUTPUT_TYPE_NONE = 'none';
+    public const OUTPUT_TYPE_DOTS = 'dots';
+    public const OUTPUT_TYPES = [
+        self::OUTPUT_TYPE_NONE,
+        self::OUTPUT_TYPE_DOTS,
+    ];
```

Comment:
I am up for having the constants/enums for those values.

Yet, I do not find it proper to have them declared in the interface.

`class FooOutput implements ProgressOutputInterface` - why would `FooOutput` be aware `OUTPUT_TYPE_DOTS` , at all? And then, when someone would add `FooOutput`, why they would need to touch the interface to add the `OUTPUT_TYPE_FOO` _to interface_ ?
and then imagine that it's non-internal class and one would like to register external , custom ProgressOutput class - and boom, they cannot adjust interface at source repo.

This should be outside of the interface and allow for extensions. Maybe mimic what we do for Fixers (and it's possibility to register custom fixers - not saying we need to introduce that possibility in this PR, but let's have a solution that will allow for that in future)

Reply:
Agree, moved to `OutputProgressType` class.

Reply:
this by no means addresses the majority of my initial comment
> And then, when someone would add FooOutput, why they would need to touch the interface to add the OUTPUT_TYPE_FOO to ~interface~ *enum* ?
and then imagine that it's non-internal class and one would like to register external , custom ProgressOutput class - and boom, they cannot adjust interface at source repo.
>
> This should be outside of the interface and allow for extensions. Maybe mimic what we do for Fixers (and it's possibility to register custom fixers - not saying we need to introduce that possibility in this PR, but let's have a solution that will allow for that in future)

Reply:
See [this comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6848#discussion_r1186573954).